libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    public class DialogContext
+    {
+        private object _finalResult;
+
+        public DialogSet Dialogs { get; set; }
+        public ITurnContext Context { get; set; }
+        public Stack<DialogInstance> Stack { get; set; }
+
+        /// <summary>
+        /// Creates a new DialogContext instance.
+        /// </summary>
+        /// <param name="dialogs">Parent dialog set.</param>
+        /// <param name="context">Context for the current turn of conversation with the user.</param>
+        /// <param name="stack">Current dialog stack.</param>
+        public DialogContext(DialogSet dialogs, ITurnContext context, Stack<DialogInstance> stack)
+        {
+            Dialogs = dialogs;
+            Context = context;
+            Stack = stack;
+        }
+
+        /// <summary>
+        /// Returns the cached instance of the active dialog on the top of the stack or `undefined` if the stack is empty.
+        /// </summary>
+        public DialogInstance Instance
+        {
+            get
+            {
+                if (Stack.Any())
+                {
+                    return Stack.Peek();
+                }
+                return null;
+            }
+        }
+
+        public DialogResult DialogResult
+        {
+            get
+            {
+                return new DialogResult
+                {
+                    Active = Stack.Count > 0,
+                    Result = _finalResult
+                };
+            }
+        }
+
+
+        /// <summary>
+        /// Pushes a new dialog onto the dialog stack.
+        /// </summary>
+        /// <param name="dialogId">ID of the dialog to start.</param>
```

Comment:
Please make these code comments useful. This is not an excercise in "How useless can we make a code comment?"

the following, for example, is not a legit code comment:
<param name="myString">The My String Parameter</param>

Reply:
These comments will **all** be re-written by our documentation writers.  There needs to be enough information for Jonathan and Cash to grok what the method is doing.